src/backend/utils/storage.js

```diff
@@ -1,25 +1,49 @@
+const crypto = require('crypto');
+const normalizeUrl = require('normalize-url');
+const { logger } = require('./logger');
 const { redis } = require('../lib/redis');
 
+const createKey = (url, prefix) => {
+  const namespace = `t:${prefix}:`;
+
+  try {
+    return namespace.concat(
+      crypto
+        .createHash('sha256')
+        .update(url)
+        .digest('base64')
+    );
+  } catch (error) {
+    logger.error(`There was an error processing ${url}`);
+    throw error;
+  }
+};
+
+const createPostKey = uri => {
+  return createKey(normalizeUrl(uri), 'post');
+};
+
+const createFeedKey = uri => {
+  return createKey(uri, 'feed');
+};
+
 // Redis Keys
-const FEED_ID = 'feed_id';
-const FEEDS = 'feeds';
+const FEEDS = 't:feeds';
 
-const POSTS = 'posts';
+const POSTS = 't:posts';
 
 module.exports = {
   addFeed: async (name, url) => {
-    // "If the key does not exist, it is set to 0 before performing the operation"
-    // https://redis.io/commands/INCR
-    const feedId = await redis.incr(FEED_ID);
+    const key = createFeedKey(url);
     await redis
       .multi()
       // Using hmset() until hset() fully supports multiple fields:
       // https://github.com/stipsan/ioredis-mock/issues/345
       // https://github.com/luin/ioredis/issues/551
-      .hmset(feedId, 'name', name, 'url', url)
-      .sadd(FEEDS, feedId)
+      .hmset(key, 'name', name, 'url', url)
+      .sadd(FEEDS, key)
       .exec();
-    return feedId;
+    return key;
```

Comment:
Question: should we return the key, or the key with the namespace removed?

Reply:
I wonder if we should return nothing here.  The only place I see us using this is in one unit test that checks the return type.  We could just remove that test.  I don't think we ever need to use the key that we get back, since we know that it's based on the URL.

Yeah, let's remove this line and the test.